util/a2e/atoe_utils.c

```diff
@@ -178,7 +174,7 @@ typedef enum {
  *               left_justify   Left justify number flag
  *               min_width      Minimum number of charaters value will
  *                              occupy
- *               precision
+ *               precision      Fix the width with leading zeros
```

Comment:
I don't think this is right. Leading zeros should only be written if `zero_pad`.

Reply:
I know it looks strange but that is how standard printf behaves! check this example from [this manual](https://linux.die.net/man/3/vsnprintf):
```
To print a date and time in the form "Sunday, July 3, 10:02", where weekday and month are pointers to strings:
#include <[stdio.h](https://linux.die.net/include/stdio.h)>
fprintf(stdout, "%s, %s %d, %.2d:%.2d\n",
        weekday, month, day, hour, min);
```
in this case, the minute was converted to `02` with `%.2d` format string. There are more examples like this in the manual!

Reply:
The behaviour of `%.2d` and `%02d` is slightly different. the `%02d` adjust the length of the whole string including the sign whereas the `%.2d` only adjust the number length excluding the sign. we can also combine those for instance `printf("%4.2d",2) ` would result `" -02"`

Reply:
Ok, so I'm still learning. Thanks.